### ISSUE TITLE
Compatibility difference between release and development builds.

### ISSUE THREAD

**Issue by atomic83GitHub** [PCSX2/pcsx2#2930]
***PCSX2 version:***
1.5dev 3044

***Description of the issue:***
Various games goes ingame with development builds while they can't pass the loading screen with release builds.

***Affected games (incomplete):***

- shaun murray pro skater

- ESPN major league baseball

- Ratchet and Clank 3 NTSC (crash in DAXX when too much effects are in use.)

***How to reproduce the issue:***
Load one of these games with booth the release and development builds and observe the issue.

***Not a regression***

***PC specifications:***
AMD FX8350 GTX 1060 Gskill 16Gb

**Comment by lightningterror** [comment]
There is a bunch of code for dev releases only, maybe some can be adjusted to be present for release builds. You'd need to know where to look tho and what the game does, are there any console logs that show anything unusual/useful?

**Comment by prafullpcsx2** [comment]
World Series Baseball 2k3 can be added to the list.

**Comment by atomic83GitHub** [comment]
In ESPN Major league baseball, Gif Unit - GS packet size exceeded VU memory size! is consistently spamming in the console.

**Comment by lightningterror** [comment]
Is there a difference between the logs Dev vs Release?

**Comment by atomic83GitHub** [comment]
Yes, these Gif Unit - GS packet size exceeded VU memory size! are not on the dev release.

**Comment by lightningterror** [comment]
Try messing with the IsDevBuild variable. Maybe you'll hit lucky.

https://github.com/PCSX2/pcsx2/blob/2e1db411fa95cf4ac6e65a345014c4753a245d96/pcsx2/Gif_Unit.h

**Comment by kozarovv** [comment]
For ESPN major league baseball, and World Series Baseball 2k3 this will be probably that code: 

https://github.com/PCSX2/pcsx2/blob/6e22efe790a99d9750b407acd4690cc8b93d34d5/pcsx2/x86/ix86-32/iR5900-32.cpp#L1057
https://github.com/PCSX2/pcsx2/blob/6e22efe790a99d9750b407acd4690cc8b93d34d5/pcsx2/x86/ix86-32/iR5900-32.cpp#L1384

As  those games use same engine as Marvel Nemesis: Rise of the Imperfects, and enabling that allow it to goes ingame. https://github.com/PCSX2/pcsx2/issues/1323#issuecomment-590815178

**Comment by kozarovv** [comment]
For Wakeboarding Unleashed Featuring Shaun Murray difference is here:

VU1 is not allowed to out of loop early (force exit) on release builds when analyze with mVUtestCycles, that's correct behavior. 
https://github.com/PCSX2/pcsx2/blob/b5625ad8b08ddfc3158d1d19aa5e616994fce993/pcsx2/x86/microVU_Compile.inl#L359

if other games from list also show message **microVU1 Warning: Exiting from Possible Infinite Loop** when running on devbuild. Then is the same thing that make difference between builds.

For Wakeboarding Unleashed Featuring Shaun Murray devbuild give better result because game really can't get out of loop without m bit handling. No idea that other games share the same code path. 

Also i was really wrong in comment that i made 25feb. I didn't noticed that refraction fixed one of COP2 instructions, and that was what allowed Marvel Nemesis: Rise of the Imperfects goes ingame. 
I just used latest code then from github, and thought that my change made difference, but it was refraction commit in mean time that fixed it. Sorry for confusion.

**Comment by MrCK1** [comment]
I would say R&C no longer crashes, the other 3 games need to be validated here


shaun murray pro skater

ESPN major league baseball

World Series Baseball 2k3

**Comment by TheTechnician27** [comment]
> I would say R&C no longer crashes, the other 3 games need to be validated here
> 
> shaun murray pro skater
> 
> ESPN major league baseball
> 
> World Series Baseball 2k3

I own World Series Baseball 2K3 and can say that I had no issues with it during a 20-ish-minute play session.

**Comment by JordanTheToaster** [comment]
Closing as this is likely no longer an issue.
